pydantic_ai_slim/pydantic_ai/_agent_graph.py

```diff
@@ -701,7 +701,7 @@ async def process_function_tools(  # noqa: C901
             output_parts.append(part)
         else:
             try:
-                result_data = await tool_manager.handle_call(call)
+                result_data = await tool_manager.handle_call(call, usage_limits=ctx.deps.usage_limits)
```

Comment:
What do you think about not passing `usage_limits` in this case, because it's an output tool, and having the presence of `usage_limits` determine whether to increment/enforce them? That way we can remove the explicit `kind == 'output'` check from the `ToolManager`